packages/console/src/command-history/CommandHistoryViewportUpdater.tsx

```diff
@@ -55,34 +55,41 @@ function CommandHistoryViewportUpdater({
     [table]
   );
 
-  useEffect(() => {
-    const cleanup = table.onUpdate(
-      (viewportData: ViewportData<CommandHistoryStorageItem>) => {
-        onUpdate({
-          items: viewportData.items ?? [],
-          offset: viewportData.offset ?? 0,
-        });
-      }
-    );
-
-    return () => {
-      log.debug('onUpdate cleanup');
-      cleanup();
-    };
-  }, [table, onUpdate]);
-
-  useEffect(() => {
-    table.setSearch(search ?? '');
-  }, [table, search]);
+  useEffect(
+    function updateTableAndReturnCleanup() {
+      const cleanup = table.onUpdate(
+        (viewportData: ViewportData<CommandHistoryStorageItem>) => {
+          onUpdate({
+            items: viewportData.items ?? [],
+            offset: viewportData.offset ?? 0,
+          });
+        }
+      );
 
-  useEffect(() => {
-    throttledUpdateViewport({
-      top,
-      bottom,
-      columns,
-    });
-  }, [throttledUpdateViewport, top, bottom, columns, search, isReversed]);
+      return () => {
+        log.debug('onUpdate cleanup');
+        cleanup();
+      };
+    },
+    [table, onUpdate]
+  );
 
+  useEffect(
+    function setSearchbar() {
+      table.setSearch(search ?? '');
+    },
```

Comment:
You're not setting the search bar here, you're setting the search query on the table. More appropriate name would be `setSearchQuery`, `setSearchText`, or even just `updateSearch`:
```suggestion
    function updateSearch() {
      table.setSearch(search ?? '');
    },
```